Minio/HttpRequestMessageBuilder.cs

```diff
@@ -56,6 +57,7 @@ public HttpRequestMessageBuilder(HttpMethod method, Uri requestUri)
 
     public Uri RequestUri { get; set; }
     public Action<Stream> ResponseWriter { get; set; }
+    public Func<Stream, Task> FunctionResponseWriter { get; set; }
```

Comment:
To make it even more async/await friendly - `Func<Stream, **CancellationToken**, Task>` and `await FunctionResponseWriter(stream, cancellationToken);` on the caller side

Reply:
@codemnk ,

Thank you. 
Yes, we can do this.
Either I'll create a sole PR for it or piggyback and include it in the upcoming PR to add 2 new bucket notification tests into the functional tests.